packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1943,6 +1952,57 @@ export function catalogServiceBuilder(
 
       return updatedEService;
     },
+    async updateAgreementApprovalPolicy(
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      seed: catalogApi.UpdateEServiceDescriptorAgreementApprovalPolicySeed,
+      { authData, correlationId, logger }: WithLogger<AppContext<UIAuthData>>
+    ): Promise<EService> {
+      assertFeatureFlagEnabled(
+        config,
+        "featureFlagAgreementApprovalPolicyUpdate"
+      );
+
+      logger.info(
+        `Updating Agreement approval policy of Descriptor ${descriptorId} for EService ${eserviceId}`
+      );
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        authData,
+        readModelService
+      );
+
+      const descriptor = retrieveDescriptor(descriptorId, eservice);
+      assertDescriptorUpdatable(descriptor);
```

Comment:
Since we're working on this, could you also update the methods `updateEServiceName` and `updateDescriptorAttributes` to reuse this `assertDescriptorUpdatable` method? 
Same as done here, or in `updateEServiceDelegationFlags`, or in `updateEServiceDescription`. 
In this way, all the methods that update a published e-service/descriptor follow the same pattern, and we can then close this old PR because not relevant anymore: https://github.com/pagopa/interop-be-monorepo/pull/1383